models/schemas/reviewSchema.ts

```diff
@@ -34,3 +36,17 @@ const ReviewSchema = new mongoose.Schema<Review>({
 });
 
 export default ReviewSchema;
+ReviewSchema.post('save', async function() {
+  const level = await LevelModel.findById(this.levelId);
+
+  if (level) {
+    await refreshIndexCalcs(level);
```

Comment:
And for reviews, what if you stored `calc_reviews_score_total` instead of `calc_reviews_score_avg`, and then we can just compute avg each time. Should make for faster updates although maybe it makes things more complicated.

So then here all you'd have to do is increment the count by 1 and the total by this.score

Reply:
Oh I see, this will cause problems for sorting by the average... In that case maybe you could store both the total and the average? Just trying to think of a way to have the least db queries.